magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BackupDiscovery.java

```diff
@@ -92,47 +95,68 @@ public void discoverPlans(ObjectMapper mapper, Session session, Region region, E
     final var RESOURCE_TYPE = BackupVault.RESOURCE_TYPE;
     try {
-      client.listBackupVaultsPaginator().stream()
-        .forEach(backupVaultsResponse -> backupVaultsResponse.backupVaultList()
+      client.listBackupVaults().backupVaultList().stream()
           .forEach(backupVault -> {
             var data = new MagpieAwsResource.MagpieAwsResourceBuilder(mapper, backupVault.backupVaultArn())
               .withResourceName(backupVault.backupVaultName())
               .withResourceId(backupVault.backupVaultName())
               .withResourceType(RESOURCE_TYPE)
-              .withConfiguration(mapper.valueToTree(backupVault.toBuilder()))
               .withCreatedIso(backupVault.creationDate())
               .withAccountId(account)
               .withAwsRegion(region.toString())
               .build();
 
-            discoverTags(client, backupVault, data);
+            AWSUtils.update(data.configuration, backupVault);
+
+            discoverVaultTags(client, backupVault, data);
 
             emitter.emit(VersionedMagpieEnvelopeProvider.create(session, List.of(fullService() + ":backupVault"), data.toJsonNode()));
-          }));
+          });
     } catch (SdkServiceException | SdkClientException ex) {
       DiscoveryExceptions.onDiscoveryException(RESOURCE_TYPE, null, region, ex);
     }
   }
 
-  private void discoverTags(BackupClient client, BackupVaultListMember resource, MagpieAwsResource data) {
+  private void discoverVaultTags(BackupClient client, BackupVaultListMember resource, MagpieAwsResource data) {
     final String keyname = "Tags";
     getAwsResponse(
-      () -> client.listTagsPaginator(ListTagsRequest.builder().resourceArn(resource.backupVaultArn()).build())
-        .stream()
-        .map(ListTagsResponse::toBuilder)
-        .collect(Collectors.toList()),
-      (resp) -> AWSUtils.update(data.supplementaryConfiguration, Map.of(keyname, resp)),
-      (noresp) -> AWSUtils.update(data.supplementaryConfiguration, Map.of(keyname, noresp))
+      () -> client.listTags(ListTagsRequest.builder().resourceArn(resource.backupVaultArn()).build()),
+      (resp) -> data.supplementaryConfiguration = AWSUtils.update(
+        data.supplementaryConfiguration, Map.of(keyname, resp)),
+      (noresp) -> data.supplementaryConfiguration = AWSUtils.update(
+        data.supplementaryConfiguration, Map.of(keyname, noresp))
+    );
+
+  }
+
+
+  private void getPlanDetails(BackupClient client, MagpieAwsResource data) {
+    getAwsResponse(
+      () -> client.getBackupPlan(GetBackupPlanRequest.builder().backupPlanId(data.resourceId).build()),
+      (resp) -> AWSUtils.update(data.configuration, resp),
+      (noresp) -> AWSUtils.update(data.supplementaryConfiguration, noresp)
     );
   }
 
+//  private void discoverTags(BackupClient client, BackupVaultListMember resource, MagpieAwsResource data) {
```

Comment:
Any reason to leave the commented block?

Reply:
None at all!